template/tests/test-xstoragebucket-kcl/main.k

```diff
@@ -20,6 +19,17 @@ of the dependent resources.
 import models.io.upbound.dev.meta.v1alpha1 as metav1alpha1
 import resources
 
+# It's a best practice for composition functions to return only fields whose
+# values they care about, since they become the owner of any field they
+# return. Allow for fields with defaults to be omitted by the composition
+# function by clearing them in our expected resources.
+_stripDefaults = lambda obj: any -> any {
+    obj | {
+        spec.deletionPolicy = Undefined
+        spec.managementPolicies = Undefined
+    }
+}
```

Comment:
Note that the KCL implementation passes tests without this, since the KCL function returns the default values. That isn't super great, but it's how the schemas work today. With this change, the tests pass for both KCL and other languages since we're not asserting anything about the fields we don't care about.